### DISCUSSION TITLE
Transfer learning

### DISCUSSION THREAD

**Discussion by cosao** [open-edge-platform/anomalib#788] [Q&A]
I got the ckpt file through model training.

But I want to proceed with more training through transfer learning.

I'd appreciate it if you could tell me how to do it in this regard.

**Comment by djdameln** [comment]
Can you elaborate a bit more on your use-case? What exactly is it that you want to achieve with transfer learning? And which model are you using?

**Comment by cosao** [comment]
> Can you elaborate a bit more on your use-case? What exactly is it that you want to achieve with transfer learning? And which model are you using?

Thank you for your answer.

I trained the Patchcore model with custom data.

However, due to the large amount of data, CUDA's OOM error occurred.

Therefore, we intend to proceed with model learning by applying transfer learning.

This is a countermeasure against OOM errors, and the final goal is to conduct model training by dividing a large data set into several parts.

**Comment by djdameln** [comment]
If I understand your use-case correctly, you would like to split up your dataset into several parts, and train PatchCore model in multiple rounds on the different subsets.

I'm afraid this will not achieve the desired effect. PatchCore models are trained by extracting feature embeddings from the training data using a pre-trained convnet with fixed weights. At the end of the training sequence, all feature embeddings are collected in a memory bank. The memory bank of the full dataset is then reduced to a coreset which is used to compare test samples.

Starting a new round of training from a previously trained model on a new part of the dataset would not work, because the memory bank obtained by the previous training round would just be overwritten by the memory bank obtained from the new data. Maybe it would be possible to achieve such a functionality with modifications to the implementation,
but this would not be straightforward, because the coreset subsampling mechanism would have to be changed. If you want to exchange ideas about how this could achieved, you could start a new thread in our [discussions section](https://github.com/openvinotoolkit/anomalib/discussions).

Unfortunately, PatchCore is a memory intensive algorithm, which may limit its usability for large datasets. Maybe you could try if some less memory-intensive algorithms, such as DRAEM or STFPM would lead to the desired results for your dataset.
